GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.UploadObject.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Upload;
+using Google.Common;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    public partial class StorageClient
+    {
+        /// <summary>
+        /// Uploads the data for an object in storage synchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket in which to store the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <returns>The metadata for the uploaded object.</returns>
+        public Object UploadObject(string bucket, string objectName, string contentType, Stream source)
+        {
+            return UploadObject(bucket, objectName, contentType, source, options: null, progress: null);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage synchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <param name="options">Additional options for the upload. May be null, in which case appropriate
+        /// defaults will be used.</param>
+        /// <param name="progress">Progress reporter for the upload. May be null.</param>
+        /// <returns>The metadata for the uploaded object.</returns>
```

Comment:
_The metadata_ can be confusing when combining it with the notion of metadata in the API itself. Maybe calling it the `Object` uploaded?